modules/ipp/scanner.go

```diff
@@ -227,47 +228,43 @@ func readAttributeFromBody(attrString string, body *[]byte) ([][]byte, error) {
 		for tag, nameLength := valueTag, int16(0); tag == valueTag && nameLength == 0; {
 			var length int16
 			if err := binary.Read(buf, binary.BigEndian, &length); err != nil {
-				//Couldn't read length of content for some reason
-				//return nil, err
-				continue
+				//Couldn't read length of content
+				return vals, err
 			}
 			val := make([]byte, length)
 			if err := binary.Read(buf, binary.BigEndian, &val); err != nil {
-				//Content not read without issue.
-				//vals = append(vals, val)
-				//return vals, err
-				continue
+				//Couldn't read content
+				vals = append(vals, val)
+				return vals, err
 			}
 			//return &val, nil
 			vals = append(vals, val)
 			if err := binary.Read(buf, binary.BigEndian, &tag); err != nil {
 				//Couldn't read next valueTag
-				//break
-				continue
+				return vals, err
 			}
 			if err := binary.Read(buf, binary.BigEndian, &nameLength); err != nil {
 				//Couldn't read next nameLength
-				//break
-				// TODO: Conclude correct error behavior for all of these blocks
-				continue
+				return vals, err
 			}
 		}
-		return vals, nil // FIXME: This should be whatever error occurred during a break case
+		return vals, nil
 	}
 	//The attribute was not present
 	return nil, errors.New("Attribute \"" + attrString + "\" not present.")
 }
 
-func versionNotSupported(resp *http.Response, scanner *Scanner) bool {
-	if resp != nil && resp.Body != nil {
-		buf := bufferFromBody(resp, scanner)
+func versionNotSupported(body *string, scanner *Scanner) bool {
```

Comment:
FYI, passing a string "by value" doesn't mean that it's going to copy the entire body -- it's sort of like Java, where it copies the length / pointer.